crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -236,10 +236,28 @@ impl EngineState {
                     // Updating existing overlay
                     for (k, v) in env.drain() {
                         if k == "config" {
-                            self.config = v.clone().into_config().unwrap_or_default();
+                            // Don't insert the record as the "config" env var as-is.
+                            // Instead, parse the config, then create a new Record holding
+                            // only the valid values, and put THAT in env_vars.
+
+                            // NOTE: currently (Dec 2022) the subrecords of config.menus, config.keybindings and config.hooks
+                            // are NOT reconstructed by config_to_nu_record(), but restored from self.config as-is.
+                            // As such, invalid values may persist on them (as into_config() doesn't remove them.
+                            if let Ok(config) = v.clone().into_config() {
```

Comment:
This `into_config` will still return a new `Config`, eventually, you will have another default config and the only change to that updated field